hack/run_in_virtualenv.sh

```diff
@@ -0,0 +1,125 @@
+#!/bin/bash
+# Script to wrap a command in a specified virtualenv and automatically set that virtualenv up to match a given spec.
+# Virtualenvs may be configured by putting the following files into the hack/python_requirements/[virtualenv name] directory.
+# All files are optional, but at least one must be present to create the virtualenv
+# Files:
+# * requirements.txt: Works as a Python requirements file, listing PyPi packages to install.
+# * virtualenv_args.sh: Specifies additional arguments to pass to the virtualenv executable when creating this virtualenv.
+# * python_executable.txt: Specifies the path to the python executable to use as the core python interpreter in the virtualenv.
+# * .env: A list of environment variables to set in the virtualenv. May interpolate any values provided directly from virtualenv itself.
+# * setup.sh: An arbitrary shell script that will run when the virtualenv is created. Usable to install any custom dependencies not available from pip.
+set -e
+
+VERBOSE_MODE=0
+
+function print_help() {
+    echo "usage: ./run_in_virtualenv.sh [-v] virtualenv_name \"command goes here\""
+    echo "    -h            Display this help message."
+    echo "    -v            Turn on verbose mode. Prints detailed information about environment setup."
+}
+
+while getopts ":vh" OPTION
+do
+    case $OPTION in
+        h )
+            print_help
+            exit 0
+            ;;
+        v ) VERBOSE_MODE=1
+            ;;
+        \? )
+            echo "Invalid option: -$OPTARG" 1>&2
+            print_help
+            exit 0
+            ;;
+    esac
+done
+
+ROOT_DIRECTORY=$(readlink -e $(dirname "$0"))
```

Comment:
This bombs out for me when testing locally, with:
```
~/src/dog-aging-ingest/hack(sjp-dspdc-1433-improve-tsv-conversion-script*) » bash -x ./run_in_virtualenv.sh dap_venv                                                                                 (docker-desktop/default)
+ set -e
+ VERBOSE_MODE=0
+ getopts :vh OPTION
+++ dirname ./run_in_virtualenv.sh
++ readlink -e .
readlink: illegal option -- e
usage: readlink [-n] [file ...]
+ ROOT_DIRECTORY=
```
maybe an macos thing?

Reply:
Ah, yeah, there's a helper I have to use the OS-specific `readlink` implementation depending on the OS running the command. Let me pull that in.

Reply:
(`readlink` does something unrelated in OSX so you need to call `greadlink` to get the GNU functionality)